cpg_pipes/pipeline/cli_opts.py

```diff
@@ -0,0 +1,239 @@
+        ),
+        click.option(
+            '--input-provider',
+            'input_provider_type',
+            type=choice_from_enum(InputProviderType),
+            callback=val_to_enum(InputProviderType),
+            default=InputProviderType.SMDB.value,
+            help=f'Source of inputs. '
+            f'For "--input-source={InputProviderType.CSV.value}", '
+            f'use --input-csv to specify a CSV file location',
+        ),
+        click.option(
+            '--input-csv',
+            'input_csv',
+            help=f'CSV file to provide with --input-provider={InputProviderType.CSV.value}',
+        ),
+        click.option(
+            '--keep-scratch/--remove-scratch',
+            'keep_scratch',
+            default=False,
+            is_flag=True,
+        ),
+        click.option('--dry-run', 'dry_run', is_flag=True),
+        click.option(
+            '--skip-samples-with-missing-input',
+            'skip_samples_with_missing_input',
+            default=False,
+            is_flag=True,
+            help='For the first (not-skipped) stage, if the input for a target does not'
+            'exist, just skip this target instead of failing. E.g. if the first'
+            'stage is Align, and sequence.meta files for a sample do not exist,'
+            'remove this sample instead of failing.',
+        ),
+        click.option(
+            '--check-intermediates/--no-check-intermediates',
+            'check_intermediates',
+            default=True,
+            is_flag=True,
+            help='Within jobs, check all in-job intermediate files for possible reuse. '
+            'If set to False, will overwrite all intermediates. ',
+        ),
+        click.option(
+            '--check-expected-outputs/--no-check-expected-outputs',
+            'check_expected_outputs',
+            default=True,
+            is_flag=True,
+            help='Before running a stage, check if its input already exists. '
+            'If it exists, submit a [reuse] job instead. '
+            'Works nicely with --previous-batch-tsv/--previous-batch-id options.',
+        ),
+        click.option(
+            '--local-dir',
+            'local_dir',
+            help='Local directory for temporary files. Usually takes a few KB. '
+            'If not provided, a temp folder will be created',
+        ),
+    ]
+
+    # Click shows options in a reverse order, so inverting the list back:
+    options = options[::-1]
+
+    # Applying decorators:
+    for opt in options:
+        function = opt(function)
+
+    # Add ability to load options from a yaml file
+    # using https://pypi.org/project/click-config-file/
+    def yaml_provider(fp, _):
+        """Load options from YAML"""
+        with open(fp) as f:
+            return yaml.load(f, Loader=yaml.SafeLoader)
+
+    function = click_config_file.configuration_option(
+        provider=yaml_provider,
+        implicit=False,
+    )(function)
+    return function
```

Comment:
```suggestion
    return click_config_file.configuration_option(
        provider=yaml_provider,
        implicit=False,
    )(function)
```

Reply:
I generally find it useful to assign function result to a variable before returning, it helps in debugging